xds/server_options.go

```diff
@@ -69,8 +71,28 @@ type ServingModeChangeArgs struct {
 //
 // # Experimental
 //
+// Notice: This API is EXPERIMENTAL but kept for backward compatibility. Use
+// `ClientPoolForTesting` to set the pool directly. It will be removed in
+// later release.
+func BootstrapContentsForTesting(bootstrapContents []byte) grpc.ServerOption {
+	err := xdsclient.DefaultPool.SetFallbackBootstrapConfig(bootstrapContents)
```

Comment:
Fallback bootstrap configuration exists for a specific purpose. It exists for cases where the grpc application is running in a non-PSM environment, and therefore they don't have bootstrap configuration. This is useful in directpath deployments where the resolver takes care of generating the bootstrap configuration and it calls this function. 

This API could look like
```golang
func BootstrapContentsForTesting(bootstrapContents []byte) grpc.ServerOption {
  // Parse the bootstrap configuration. 
  // If parsing fails, throw an error log here and set `o.clientPoolForTesting` to `nil`, and return.
  
  // Create a new pool with the parsed configuration.
  // Set `o.clientPoolForTesting` to the newly created pool, and return
}
```

With the above approach, we won't need `ClientPoolForTesting`.

What do you think?

Reply:
> // Parse the bootstrap configuration. 
  // If parsing fails, throw an error log here and set `o.clientPoolForTesting` to `nil`, and return.
  
Changed to this

Reply:
> With the above approach, we won't need ClientPoolForTesting

Actually, we still need it because some tests need to keep track of pool to fetch the client